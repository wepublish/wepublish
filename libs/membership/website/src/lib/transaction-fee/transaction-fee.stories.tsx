import {css} from '@emotion/react'
import {action} from '@storybook/addon-actions'
import {Meta, StoryObj} from '@storybook/react'
import {useState} from 'react'
import {TransactionFee} from './transaction-fee'

export default {
  component: TransactionFee,
  title: 'Components/TransactionFee',
  render: function ControlledTransactionFee(args) {
    const [value, setValue] = useState(args.value)

    return (
      <TransactionFee
        {...args}
        value={value}
        onChange={periodicity => {
          args.onChange(periodicity)
          setValue(periodicity)
        }}
      />
    )
  }
} as Meta<typeof TransactionFee>

export const Default: StoryObj<typeof TransactionFee> = {
  args: {
    onChange: action('onChange')
  }
}

export const WithCustomText: StoryObj<typeof TransactionFee> = {
  ...Default,
  args: {
    ...Default.args,
    text: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.'
  }
}

export const WithClassName: StoryObj<typeof TransactionFee> = {
  ...Default,
  args: {
    ...Default.args,
    className: 'extra-classname'
  }
}

export const WithEmotion: StoryObj<typeof TransactionFee> = {
  ...Default,
  args: {
    ...Default.args,
    css: css`
      background-color: #eee;
    `
  }
}
