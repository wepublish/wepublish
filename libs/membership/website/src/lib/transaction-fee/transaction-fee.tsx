import {Checkbox, darken, FormControlLabel, lighten, styled} from '@mui/material'
import {BuilderTransactionFeeProps, useWebsiteBuilder} from '@wepublish/website/builder'
import {forwardRef, useId} from 'react'
import {MdFavorite} from 'react-icons/md'

const iconSize = 32

export const TransactionFeeWrapper = styled(FormControlLabel)`
  justify-self: center;
  max-width: 75ch;
  margin: 0;
  position: relative;
  padding: ${({theme}) => `
    calc(${iconSize / 2}px + ${theme.spacing(1)})
    ${theme.spacing(2)}
    ${theme.spacing(2)}
  `};
  background-color: ${({theme}) => lighten(theme.palette.error.main, 0.9)};
  color: ${({theme}) => darken(theme.palette.error.main, 0.6)};
  border-radius: ${({theme}) => theme.shape.borderRadius}px;
`

export const TransactionFeeIcon = styled('div')`
  display: inline-flex;
  background-color: ${({theme}) => lighten(theme.palette.error.main, 0.9)};
  color: ${({theme}) => theme.palette.error.main};
  font-size: ${iconSize}px;
  padding: ${({theme}) => theme.spacing(1)};
  border-radius: 50%;
  position: absolute;
  top: 0;
  left: 50%;
  transform: translate3d(-50%, -50%, 0);
`

export const TransactionFee = forwardRef<HTMLButtonElement, BuilderTransactionFeeProps>(
  function TransactionFee({text, onChange, value, className, name}, ref) {
    const id = useId()
    const {
      meta: {siteTitle}
    } = useWebsiteBuilder()

    return (
      <TransactionFeeWrapper
        className={className}
        control={
          <>
            <TransactionFeeIcon>
              <MdFavorite />
            </TransactionFeeIcon>

            <Checkbox
              id={id}
              name={name}
              ref={ref}
              checked={value ?? false}
              onChange={(_, checked) => onChange(checked)}
              color="error"
            />
          </>
        }
        label={
          text ??
          `Ja, ich will die Bearbeitungsgeb체hren von 2% 체bernehmen und damit ${siteTitle} zus채tzlich unterst체tzen.`
        }
      />
    )
  }
)
