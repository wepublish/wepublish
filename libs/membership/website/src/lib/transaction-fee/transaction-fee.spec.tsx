import { render } from '@testing-library/react';
import * as stories from './transaction-fee.stories';
import { composeStories } from '@storybook/react';

const storiesCmp = composeStories(stories);

describe('TransactionFee', () => {
  Object.entries(storiesCmp).forEach(([story, Component]) => {
    it(`should render ${story}`, async () => {
      render(<Component />);
    });
  });
});
