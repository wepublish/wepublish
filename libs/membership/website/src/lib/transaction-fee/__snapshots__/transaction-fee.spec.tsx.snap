// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionFee should render Default 1`] = `
<DocumentFragment>
  <label
    class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd emotion-0 emotion-1"
  >
    <div
      class="emotion-2 emotion-3"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="m12 21.35-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
        />
      </svg>
    </div>
    <span
      class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorError MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorError MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorError MuiCheckbox-sizeMedium emotion-4"
    >
      <input
        class="PrivateSwitchBase-input emotion-5"
        data-indeterminate="false"
        id=":r0:"
        type="checkbox"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-6"
        data-testid="CheckBoxOutlineBlankIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root emotion-7"
      />
    </span>
    <span
      class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label emotion-8"
    >
      Ja, ich will die Bearbeitungsgebühren von 2% übernehmen und damit Newsroom Name zusätzlich unterstützen.
    </span>
  </label>
</DocumentFragment>
`;

exports[`TransactionFee should render WithClassName 1`] = `
<DocumentFragment>
  <label
    class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd extra-classname emotion-0 emotion-1"
  >
    <div
      class="emotion-2 emotion-3"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="m12 21.35-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
        />
      </svg>
    </div>
    <span
      class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorError MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorError MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorError MuiCheckbox-sizeMedium emotion-4"
    >
      <input
        class="PrivateSwitchBase-input emotion-5"
        data-indeterminate="false"
        id=":r1:"
        type="checkbox"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-6"
        data-testid="CheckBoxOutlineBlankIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root emotion-7"
      />
    </span>
    <span
      class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label emotion-8"
    >
      Ja, ich will die Bearbeitungsgebühren von 2% übernehmen und damit Newsroom Name zusätzlich unterstützen.
    </span>
  </label>
</DocumentFragment>
`;

exports[`TransactionFee should render WithCustomText 1`] = `
<DocumentFragment>
  <label
    class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd emotion-0 emotion-1"
  >
    <div
      class="emotion-2 emotion-3"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="m12 21.35-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
        />
      </svg>
    </div>
    <span
      class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorError MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorError MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorError MuiCheckbox-sizeMedium emotion-4"
    >
      <input
        class="PrivateSwitchBase-input emotion-5"
        data-indeterminate="false"
        id=":r2:"
        type="checkbox"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-6"
        data-testid="CheckBoxOutlineBlankIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root emotion-7"
      />
    </span>
    <span
      class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label emotion-8"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
    </span>
  </label>
</DocumentFragment>
`;

exports[`TransactionFee should render WithEmotion 1`] = `
<DocumentFragment>
  <label
    class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd emotion-0 emotion-1"
  >
    <div
      class="emotion-2 emotion-3"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="m12 21.35-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
        />
      </svg>
    </div>
    <span
      class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorError MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorError MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorError MuiCheckbox-sizeMedium emotion-4"
    >
      <input
        class="PrivateSwitchBase-input emotion-5"
        data-indeterminate="false"
        id=":r3:"
        type="checkbox"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-6"
        data-testid="CheckBoxOutlineBlankIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root emotion-7"
      />
    </span>
    <span
      class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label emotion-8"
    >
      Ja, ich will die Bearbeitungsgebühren von 2% übernehmen und damit Newsroom Name zusätzlich unterstützen.
    </span>
  </label>
</DocumentFragment>
`;
